src/nvim/api/vim.c

```diff
@@ -1964,9 +1952,7 @@ Array nvim__inspect_cell(Integer grid, Integer row, Integer col, Arena *arena, E
     if (wp != NULL && wp->w_grid_alloc.chars != NULL) {
       g = &wp->w_grid_alloc;
     } else {
-      api_set_error(err, kErrorTypeValidation,
-                    "No grid with the given handle");
-      return ret;
+      VALIDATE_INT(true, "grid handle", grid, return ret);
```

Comment:
This style will hurt treesitter. Best try and phrase the macros like:

```c
VALIDATE_INT(true, "grid handle", grid) {
   return ret;
};
```

Reply:
yeah that avoids the non-obvious control flow too.